### DISCUSSION TITLE
SSG: pagination, filtering, search, localization

### DISCUSSION THREAD

**Discussion by Sendturn** [vercel/next.js#41785] [Help]
We are planning to start a project for a SEO-driven website that will contain product reviews. Next.JS seems to be a top pick due to its SSR/SSG capabilities as well as React nativeness. However, the main discussion now happens around should we consider SSR or SSG for our pages that will contain product lists and use following features:
1. Pagination
2. Sorting
3. Filtering
4. Localization (default product filtering according to visitor country)
5. Search

So far we have found out that we can implement pagination and localization using SSG - just generate static pages for separate product pages and countries, but it seems that for sorting, filtering and search we either need to use SSG + client-side data fetching or SSR. Unfortunately, SSG + CSDF might jeopardize SEO, which is very important for us, so the only solution seems to go with SSR.

Do you agree with us?

**Comment by wmlutz** [comment]
I've been looking for a good blog post or guide that goes into these details. I'd love to learn about some best practices and design patterns that work well.
